sw/device/tests/alert_handler_lpg_reset_toggle.c

```diff
@@ -0,0 +1,548 @@
+       .signal = 0,
+       .duration_cycles = 2000}};
+
+  dif_alert_handler_class_config_t class_config = {
+      .auto_lock_accumulation_counter = kDifToggleDisabled,
+      .accumulator_threshold = 0,
+      .irq_deadline_cycles = 10000,
+      .escalation_phases = esc_phases,
+      .escalation_phases_len = ARRAYSIZE(esc_phases),
+      .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase1,
+  };
+
+  dif_alert_handler_class_config_t class_configs[] = {class_config,
+                                                      class_config};
+
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA,
+                                         kDifAlertHandlerClassB};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = array_idx,
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = ping_timeout,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));
+}
+
+/**
+ * A utility function to wait enough until the alert handler pings a peripheral
+ * alert
+ */
+void wait_enough_for_alert_ping() {
+  // wait enough
+  if (kDeviceType == kDeviceFpgaCw310) {
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**16)*(400ns) = 12.8s
+    busy_spin_micros(1000 * 12800);
+  } else if (kDeviceType == kDeviceSimDV) {
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**3)*(40ns) = 160us
+    busy_spin_micros(160);
+  } else {
+    // Verilator
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**16)*(8us) = 256s
+    // This seems to be impractical for the current clock frequency config
+    // of the Verilator tests (kClockFreqPeripheralHz = 125K).
+    LOG_FATAL("SUPPORTED PLATFORMS: DV and FPGA");
+    LOG_FATAL("TO SUPPORT THE PLATFORM %d, COMPUTE THE RIGHT WAIT-TIME",
+              kDeviceType);
+  }
+}
+
+/**
+ * Get `num` distinct random numbers in the range [0, `max`] from
+ * RV_CORE_IBEX_RND_DATA.
+ *
+ * @param ibex The Ibex DIF object.
+ * @param num The number of random numbers to get.
+ * @param[out] rnd_buf Pointer to the buffer to write the random numbers to.
+ * @param max The maximum random value returned.
+ */
+static void get_rand_words(dif_rv_core_ibex_t *ibex, int num, uint32_t *rnd_buf,
+                           uint32_t max) {
```

Comment:
Adding on to what @cindychip said, I think I remember this function from another test as well. But I agree it's fine to move these in a follow-up PR.

Reply:
Yep, actually, I think I took this from one of the tests you authored :) I will check if there is a common area for this one too.